neptune-core/src/application/json_rpc/README.md

```diff
@@ -0,0 +1,43 @@
+# How to Add a New RPC Method
+
+Let’s say you want to add a new method called `submit_block`.
+
+## Core
+
+1. **Add a new operation**
+
+   * In `json_rpc::core::api::ops::RpcApiOps`, create a new variant (e.g. `SubmitBlock`).
+
+2. **Create request and response types**
+
+   * In `json_rpc::core::model::message`, define a matching pair of structs:
+     `SubmitBlockRequest` and `SubmitBlockResponse`.
+
+3. **Update the RPC API**
+
+   * In `json_rpc::core::api::rpc::RpcApi`, add two async functions:
+
+     ```rust
+     async fn submit_block(
+         &self,
+         block: RpcBlock,
+     ) -> RpcResult<SubmitBlockResponse>;
+
+     async fn submit_block_call(
+         &self,
+         request: SubmitBlockRequest
+     ) -> RpcResult<SubmitBlockResponse>;
+     ```
+   * The first function should just call the second one.
+
+---
+
+## Server
+
+1. **Implement the `_call` method**
+
+   * In `json_rpc::server::service::RpcApi`, implement the function with the `_call` suffix.
```

Comment:
I'm a bit confused by this. Are you adding two functions to the trait? Wouldn't you have to implement both functions then? Or is there some macro magic that treats function names ending with "_call" in a special way?